src/centengine/engine_impl.cc

```diff
@@ -939,18 +943,23 @@ grpc::Status engine_impl::ScheduleHostDowntime(
       temp_host = it->second;
     if (temp_host == nullptr)
       return 1;
-    if (request->author().empty() || request->comment_data().empty())
+    if (request->author().empty() || request->comment_data().empty() ||
+        !request->has_start() || !request->has_end() ||
+        !request->has_triggered_by() || !request->has_duration() ||
+        !request->has_entry_time())
```

Comment:
Correct me if I'm wrong...

Here, we are in the packaged_task. This will be executed by the command_manager a little time later.
And I have the feeling you are just checking errors as for host name. If one of those fields is not filled, the packaged task returns 1 to raise an error.

So my question, why don't make the check just after the first about the host name and without waiting for the packaged task? Moreover, you could return a message about the error instead of just return an OK status.

Reply:
After thinking, if those values are mandatory, don't use googlt.protobuf encapsulations but just the base types.
If they are not filled by the user, they will be considered as 0 or empty. And you can keep your test to check those values.